spec/reek/configuration/configuration_file_finder_spec.rb

```diff
@@ -1,34 +1,58 @@
-require 'spec_helper'
-require 'reek/cli/application'
+# encoding: UTF-8
 
-include Reek::Cli
-include Reek::Configuration
+require 'pathname'
+require 'tmpdir'
+require 'spec_helper'
 
 describe ConfigurationFileFinder do
   describe '.find' do
-    let(:sample_config_path) { Pathname.new('spec/samples/simple_configuration.reek') }
-    let(:config_path_same_level) { Pathname.new('spec/samples/simple_configuration.reek') }
-    let(:options_with_config_file) { double(config_file: sample_config_path) }
-    let(:options_without_config_file) { double(config_file: nil) }
+    it 'returns the config_file if it’s set' do
+      config_file = double
+      found = ConfigurationFileFinder.find(double(config_file: config_file))
+      expect(found).to eq(config_file)
+    end
 
-    it 'should return the config file we passed as cli option if given' do
-      expect(ConfigurationFileFinder.find(options_with_config_file)).to eq(sample_config_path)
+    it 'returns the file in current dir if config_file is nil' do
+      options = double(config_file: nil)
+      current = Pathname.new('spec/samples')
+      found = ConfigurationFileFinder.find(options, current: current)
+      expect(found).to eq(Pathname.new('spec/samples/exceptions.reek'))
     end
 
-    it 'should return the first configuration file it can find in the current directory' do
-      allow(ConfigurationFileFinder).to receive(:detect_or_traverse_up).
-        and_return config_path_same_level
+    it 'returns the file in current dir if options is nil' do
+      current = Pathname.new('spec/samples')
+      found = ConfigurationFileFinder.find(nil, current: current)
```

Comment:
I did that, but I don’t really like it very much – `options` is different from `current` and `home`, as `current` and `home` are dependency injections for testing, while `options` is actually used when calling by working code (the `ConfigurationFileFinder.find(nil)` calls are only in the tests, they wouldn’t happen in working code).

What I think is more problematic is that `options` can be `nil` in the first place (this is so that we can call [`AppConfiguration#initialize_with(nil)`](https://github.com/troessner/reek/blob/7ad005afdbc99129b67b414f8e6a85cd6013f8ec/lib/reek/configuration/app_configuration.rb#L16-L21)). I thought about this a bit and I think eventually we should try to simplify the configuration management – currently there’s `Cli::Options`, `Configuration::AppConfiguration`, `Cli::Input`, the Rake task… We could try coming up with a single place to store the settings and that place could then be either told by the CLI that the file was given or look for it via `ConfigurationFileFinder`.

But I need to think about this a bit more first and that’s for a quite separate PR – let me know whether this one is ok for now. :)